### ISSUE TITLE
Database disconnection acquired. Context deadline exceeded.

### ISSUE THREAD

**Issue by catroot** [sorintlab/stolon#588]
Database disconnection acquired... the proxy logs tells that:
```
2018-11-26T19:43:51.698Z	INFO	cmd/proxy.go:302	check function error	{"error": "failed to update proxyInfo: context deadline exceeded"}
....
2018-11-26T19:44:16.407Z	INFO	cmd/proxy.go:249	not proxying to master address since we aren't in the enabled proxies list	{"address": "10.0.2.65:5432"},

```
Any way to handle context deadline?

**Comment by sgotti** [comment]
@catroot this is normal if your underlying store (etcd) cannot answer to request (usually overloaded). You should check why this is happening.

**Comment by sgotti** [comment]
@catroot currently it's hardcoded to 5 seconds and the proxy will timeout in 15 second. Which is IMHO already quite a lot. We'll happily accept PR to make it configurable.

**Comment by catroot** [comment]
Yes, please. I`m using stolon behind docker and application does a heavy transactions, (db dumping for example) and kicks off the connection due to this parameter.
For my sandbox i have all of them run on the one physical machine and heavy load hangs other things.

**Comment by trmfreitas** [comment]
@catroot I am doing this change, also had the same problem you did

**Comment by sgotti** [comment]
Configurable proxyCheckInterval` and `proxyTimeout` have been implemented in #756. Etcd store timeout is hardcoded to 5 second and a store taking more than 5 second to answer is really slow and should be fixed.
